src/main/java/com/commercetools/sync/producttypes/ProductTypeSync.java

```diff
@@ -223,7 +225,16 @@ private void handleError(@Nonnull final String errorMessage, @Nullable final Thr
                 removeAndKeepTrackOfMissingNestedAttributes(newProductType, keyToIdCache))
             .map(draftWithoutMissingRefAttrs -> referenceResolver
                 .resolveReferences(draftWithoutMissingRefAttrs)
-                .thenCompose(resolvedDraft -> syncDraft(oldProductTypeMap, resolvedDraft)))
+                .thenCompose(resolvedDraft -> syncDraft(oldProductTypeMap, resolvedDraft))
+                .exceptionally(completionException -> {
+                    final String errorMessage = format(FAILED_TO_RESOLVE_REFERENCES,
```

Comment:
Why do you assume that there will be a reference resolution exception only? What about `syncDraft` method which might throw a different type of error (400 or 409 for example)?

Reply:
Since our approach now is to pass the raw exception to handleError(..) no matter which type it is, the program would not distinguish exception type inside exceptionally(..). How about re-defining the error message content not to be resolve references specific?

Reply:
> Since our approach now is to pass the raw exception to handleError(..) no matter which type it is, the program would not distinguish exception type inside exceptionally

The new approach is better but the error message is misleading if the root cause is not related to reference resolution.

> How about re-defining the error message content not to be ReferenceResolution-specific?

Please make a suggestion

Reply:
> Please make a suggestion

Because the entire process is inovked inside ProductTypeSync.syncBatch(..), we can simply change as below 

`private static final String FAILED_TO_RUN_SYNC_BATCH = "Failed to run sync batch process on productTypeDraft with key:'%s'. Reason: %s";`

The error message contains reason part, which is enough to show the actual cause. It would be like following

_Failed to run sync batch process productTypeDraft with key:'001'. Reason: Failed to resolve references on attribute definition with name 'xxx'_

It can also apply in case runtime exception is thrown